biz.aQute.bndlib/src/aQute/bnd/osgi/Macro.java

```diff
@@ -2148,4 +2151,57 @@ public Object get(String key) throws Exception {
 		});
 	}
 
+	public Map<String, String> getCommands() {
+		Set<Object> targets = new LinkedHashSet<>();
+		targets.addAll(Arrays.asList(targets));
+		Processor rover = domain;
+		while (rover != null) {
+			targets.add(rover);
+			rover = rover.getParent();
+		}
+		targets.add(this);
+		TreeMap<String, String> result = new TreeMap<>();
+
+		targets.stream()
+			.map(object -> object.getClass())
+			.flatMap(c -> Stream.of(c.getMethods()))
+			.filter(m -> !Modifier.isStatic(m.getModifiers()) && Modifier.isPublic(m.getModifiers()) && m.getName()
+				.startsWith("_"))
+			.map(this::toEntry)
+			.forEach(e -> result.put(e.getKey(), e.getValue()));
+
+		return result;
+	}
+
+	private Map.Entry<String, String> toEntry(Method m) {
+		String name = m.getName()
+			.substring(1);
+		String help;
+
+		try {
+			Class<?> c = m.getDeclaringClass();
+			Field field = c.getField(m.getName() + "Help");
+			help = (String) field.get(null);
+		} catch (Exception e) {
+			help = "";
+		}
+		String h = help;
+		return new Map.Entry<String, String>() {
```

Comment:
You know about https://docs.oracle.com/javase/8/docs/api/java/util/AbstractMap.SimpleImmutableEntry.html ? You don't need to write your own Map.Entry implementations.